File: 53/course.java
import java.util.*;
public class course{

    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter no of course: ");

        int n=sc.nextInt();

        if(n<0 || n>20){
            System.out.print("“Invalid Range");
            return;
        }

        sc.nextLine();

        System.out.print("Enter course names: ");

        String[] courses = sc.nextLine().split(" ");

        if(courses.length != n){
            System.out.println("Invalid Range");
            return;
        }

        System.out.print("Enter the course to be searched: ");

        String search = sc.next();
        boolean found = false;

        for(String course: courses){
            if(course.equalsIgnoreCase(search)){
                found=true;
                break;
            }
        }

        if(found) System.out.println(search + " course is available");
        else System.out.println(search + " course is not available");
    }
}

File: 53/course.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {

    cout << "Enter no of course: ";

    int n;
    cin >> n;

    if (n < 0 || n > 20) {
        cout << "Invalid Range";
        return 0;
    }

    cin.ignore();

    cout << "Enter course names: ";

    string line;
    getline(cin, line);

    stringstream ss(line);

    vector<string> courses;
    string course;

    while (ss >> course) {
        courses.push_back(course);
    }

    if (courses.size() != n) {
        cout << "Invalid Range";
        return 0;
    }

    cout << "Enter the course to be searched: ";

    string search;
    cin >> search;

    bool found = false;

    for (string course : courses) {

        if (course.size() == search.size()) {

            bool same = true;

            for (int i = 0; i < course.size(); i++) {

                if (tolower(course[i]) != tolower(search[i])) {
                    same = false;
                    break;
                }
            }

            if (same) {
                found = true;
                break;
            }
        }
    }

    if (found)
        cout << search << " course is available";
    else
        cout << search << " course is not available";

    return 0;
}